32967 modernize put/delete tag endpoints

hassandotcms opened the pull request:
### Proposed Changes
PUT /api/v2/tags/{idOrName} implementation:
    - Dual identification (UUID or name+siteId)
    - Name and site assignment updates
    - Conflict for duplicates
    - Validation matching POST (create tag)

This PR fixes: #32967

This PR fixes: #32967

hassandotcms committed:
refactor(tags): modernize tag API v2 endpoints for REST compliance

- Move single tag creation from POST /api/v2/tags/single to POST /api/v2/tags
- Move bulk tag creation from POST /api/v2/tags to POST /api/v2/tags/_bulk
- Add SingleTagForm with Bean Validation for single tag requests
- Add ResponseEntityRestTagView for type-safe single tag responses
- Implement SYSTEM_HOST fallback for cross-site tag retrieval
- Add comprehensive OpenAPI annotations for all endpoints
- Add custom validation for business rules (comma check, whitespace)
- Update Postman tests for new endpoint paths and response formats
- Fix test expectations to match actual API behavior (tagStorage, validation)

This refactoring aligns the Tags API with REST best practices where the base
resource path handles single entity operations and special suffixes like /_bulk
handle batch operations. Includes proper error handling, validation, and
cross-site tag discovery through SYSTEM_HOST fallback.

Refs: #32966

hassandotcms committed:
feat(tags): modernize PUT/DELETE endpoints for v2 Tags API

- Add comprehensive PUT /api/v2/tags/{idOrName} endpoint with dual identification support
- Add proper DELETE /api/v2/tags/{tagId} endpoint returning 204 No Content
- Update UpdateTagForm with @JsonAlias to support modern field names (name/tagName)
- Add strict site validation without fallback for PUT operations
- Include comprehensive OpenAPI documentation with proper error responses
- Update Postman tests for new endpoint structure and validation

hassandotcms committed:
modernize(tags): modernize create/get tags api

Make create tag idempotent, will return tag if already existing (201)

Refs: #32966

hassandotcms committed:
modernize(tags): modernize create/get tags api

move validation to a single place. be consistent where we do validations.
ref: #32966

hassandotcms committed:
modernize(tags): modernize create/get tags api

 use ResponseEntityView for proper and consistent response of errors and exceptions.
 ref: #32966

hassandotcms committed:
modernize(tags): modernize create/get tags api

 cleanup. remove extra comments.
 ref: #32966

hassandotcms committed:
modernize(tags): modernize create/get tags api

fix postman tests, and verify assertions.
ref: #32966

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 32966-modernize-post-create-tag-apis

hassandotcms committed:
modernize(tags): modernize create/get tags api

 update category resource postman test to use correct tags api.
 ref: #32966

hassandotcms committed:
modernize(tags): 32966-modernize-create-get-tag-apis

removed explicit exception catches. add debug statements.
Ref: #32966

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 32966-modernize-post-create-tag-apis

hassandotcms committed:
Merge branch '32966-modernize-post-create-tag-apis' into 32941-implement-put-delete-tag-endpoints

hassandotcms committed:
modernize(tags): put/delete tag apis

revert to original delete api.
ref: #32967

hassandotcms committed:
modernize(tags): modernize DELETE and UPDATE APIs

  - Revert DELETE to throw DoesNotExistException instead of manual Response building
  - Refactor UPDATE to use form validation and exception throwing
  - Update Postman tests to match new response formats
 ref: #32967

hassandotcms committed:
modernize(tags): put/delete tag apis

 added comments.
 ref: #32967

hassandotcms committed:
modernize(tags): put/delete tag apis

return view instead of Response.
ref: #32967

hassandotcms committed:
modernize(tags): modernize create/get tags apis

fix typo.
ref: #32966

hassandotcms committed:
modernize(tags): unify single and bulk tag creation into list-based API

  - Replace separate endpoints with unified POST /api/v2/tags
  - Fix validation errors to return consistent ErrorEntity array format
  - Add proper Swagger documentation with ResponseEntityRestTagListView
  - Update all Postman tests to match new API behavior
ref: #32967

hassandotcms committed:
Merge branch '32966-create-tags-multiple' into 32967-modernize-put-delete-tag-endpoints

hassandotcms committed:
modernize(tags): unify single and bulk tag creation into list-based API

 updated api docs.
 ref: #32967

hassandotcms committed:
modernize(tags): unify single and bulk tag creation into list-based API

remove wildcard imports.
ref: #32967

hassandotcms committed:
modernize(tags): unify single and bulk tag creation into list-based API

 use responseviewlisttagsview in response entity.
 ref: #32967

hassandotcms commented on dotCMS/src/main/java/com/dotcms/rest/api/v2/tags/TagResource.java:
```diff
@@ -162,6 +199,84 @@ public ResponseEntityTagMapView addTag(
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response createSingleTag(
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
```
done. consolidated api for single an multiple tags creation.

jdotcms replied:
I do agree the bulk should not exist, and the default add should be a collection, if want just add one, send just one

jdotcms commented on dotCMS/src/main/java/com/dotcms/rest/tag/SingleTagForm.java:
```diff
@@ -0,0 +1,65 @@
+        // Then add custom business rules
+        if (name != null) {
+            // Check for commas
+            if (name.contains(",")) {
```
StringPool.COMMA

hassandotcms committed:
modernize(tags): unify single and bulk tag creation into list-based API

fix postman failing test in content resource.
ref: #32967

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 32967-modernize-put-delete-tag-endpoints

hassandotcms committed:
modernize(tags): tags api

  - Create TagValidationHelper for DRY validation logic
  - Move SingleTagForm to v2 package with shared validation
  - Add v2-specific UpdateTagForm with consistent validation
  - Use StringPool.COMMA for consistency
ref: #32967

hassandotcms committed:
modernize(tags): tags api

refactoring. remove singletagform.
ref: #3i2967

hassandotcms committed:
modernize(tags): tags api

keep v1 updatesingleform intact
ref: #3i2967

hassandotcms committed:
Merge branch 'main' into 32967-modernize-put-delete-tag-endpoints
